distributed/scheduler.py

```diff
@@ -2733,34 +2749,43 @@ def remove_resources(self, worker):
             for resource, quantity in self.worker_resources.pop(worker).items():
                 del self.resources[resource][worker]
 
-    def coerce_address(self, addr):
+    def coerce_address(self, addr, resolve=True):
         """
-        Coerce possible input addresses to canonical form
+        Coerce possible input addresses to canonical form.
+        *resolve* can be disabled for testing with fake hostnames.
 
-        Handles lists, strings, bytes, tuples, or aliases
+        Handles strings, tuples, or aliases.
         """
-        if isinstance(addr, list):
-            addr = tuple(addr)
+        # XXX how many address-parsing routines do we have?
```

Comment:
Yeah, sorry about this.  There is some history here that is no longer relevant.  I also wouldn't mind reducing the kinds of inputs this takes.